README.md

```diff
@@ -28,126 +36,136 @@ Error cat was designed to be as easy as possible to use with express. Here is an
 
-#### {boolean} `canUseRollbar()`
-Determines whether or not ErrorCat can use Rollbar to report errors. This method
-will return `true` if, and only if, `process.env.NODE_ENV !== 'test'` and
-`process.env.ROLLBAR_KEY` is defined.
+#### Extending Error Classes
+Each of the error types, detailed below, can be extended for your own application.
+Building a robust error zoology is key in correctly implmenting a `try-catch` based
+error handling strategy in your application.
 
-Note: Error cat uses the [loadenv](http://www.npmjs.com/package/loadenv) package
-to load the environment so feel free to use its conventions to define
-`ROLLBAR_KEY`.
+Error-cat makes this easy, here's an example of how you can extend the core
+base error class to automatically increment an error counter in data-dog (via
+[monitor-dog](https://github.com/runnable/monitor-dog)) during construction:
 
-#### {Boom~Error} `create(code, message, data)`
-Creates and automatically logs a new [boom](https://www.npmjs.com/package/boom)
-via the `ErrorCat.log` method (see below). The parameters mirror `Boom.create`.
+```js
+'use strict'
+
+var BaseError = require('error-cat').BaseError
+var monitor = require('monitor-dog')
+
+/**
+ * Base class for all errors that should be monitored in data-dog.
+ * @param  {[type]} message Message for the error.
+ * @param  {[type]} data    Custom data to report to rollbar.
+ */
+var MonitoredError = function MonitoredError(message, data) {
+  BaseError.call(message, data)
+  Mon
+}
+util.inherits(MonitoredError, BaseError)
 
-#### {Boom~Error} `createAndReport(code, message, data)`
-Creates a new [boom](https://www.npmjs.com/package/boom) error, logs it via the
-`ErrorCat.log` method (see below), and reports it to rollbar via the
-`ErrorCat.report` method (see below).
+/**
+ * Monitored Error Class
+ * @module my-application:error
+ */
+module.exports = MonitoredError
+```
 
-#### {Boom~Error} `wrap(err, code, message)`
-Creates a new [boom](https://www.npmjs.com/package/boom) error from an existing
-error, and logs it via the `ErrorCat.log` method (see below). Data can be
-supplied to the resulting boom error by setting the `.data` attribute on the
-error before wrapping it. Here's an example:
+#### class `BaseError(message, data)` extends `Error`
 
 ```js
-var reqBody = 'Information that may cause request to fail.';
-var options = {
-  url: 'http://api.service.com/route',
-  body: reqBody
-};
-
-// Send an http request to an external service...
-request.post(options, function (err) {
-  if (err) {
-    // Add some data about the state that caused the error
-    err.data = { body: reqBody };
-
-    // And then wrap the error so it can be logged appropriately
-    ErrorCat.wrap(err, 502);
-  }
-});
+var ErrorCat = require('error-cat')
+
+// Report an error as a warning with the given custom data
+var myError = new ErrorCat.BaseError('Something wicked...', {
+  customRollbarField: 'customValue'
+})
+myError.level = 'warn'
```

Comment:
would be interesting if it would be possible to pass `level` and other parameters into the constructor. 
In a lot of cases I don't want to create separate var and set that level on a new line.